big_tests/tests/extdisco_SUITE.erl

```diff
@@ -0,0 +1,368 @@
+        escalus:assert(is_iq_error, [Iq], Result)
+    end,
+    escalus:fresh_story(Config, [{alice, 1}], Test).
+
+external_services_configured_incorrect_request_no_returned(Config) ->
+    % Given external service discovery is configured
+    Test = fun(Alice) ->
+
+        % When sending request with incorrect elements
+        Iq = request_external_services_incorrect(domain()),
+        escalus_client:send(Alice, Iq),
+
+        % Then iq_error is returned
+        Result = escalus_client:wait_for_stanza(Alice),
+        escalus:assert(is_iq_error, [Iq], Result)
+    end,
+    escalus:fresh_story(Config, [{alice, 1}], Test).
+
+external_service_required_elements_configured(Config) ->
+    % Given external service discovery is configured only with required elements
+    Test = fun(Alice) ->
+
+        % When requesting for external services
+        Iq = request_external_services(domain()),
+        escalus_client:send(Alice, Iq),
+
+        % Then list of external services with containing all
+        % required_elements() is returned
+        Result = escalus_client:wait_for_stanza(Alice),
+        escalus:assert(is_iq_result, [Iq], Result),
+        ?assertEqual(true, has_subelement_with_ns(Result, <<"services">>, ?NS_EXTDISCO)),
+        [required_services_are_returned(Service) || Service <- get_service_element(Result)]
+    end,
+    escalus:fresh_story(Config, [{alice, 1}], Test).
+
+%%-----------------------------------------------------------------
+%% Helpers
+%%-----------------------------------------------------------------
+
+stun_service() ->
+    {stun, [
+       {host, "1.1.1.1"},
+       {port, "3478"},
+       {transport, "udp"},
+       {username, "username"},
+       {password, "secret"}
+    ]}.
+
+turn_service() ->
+    {turn, [
+        {host, "2.2.2.2"},
+        {port, "3478"},
+        {transport, "tcp"},
+        {username, "username"},
+        {password, "secret"}
+    ]}.
+
+domain() ->
+    ct:get_config({hosts, mim, domain}).
+
+set_external_services(Opts, Config) ->
+    Module = [{mod_extdisco, Opts}],
+    ok = dynamic_modules:ensure_modules(domain(), Module),
+    Config.
+
+remove_external_services(Config) ->
+    OldModules = rpc(mim(), gen_mod, loaded_modules_with_opts, [domain()]),
+    NewModules = lists:keydelete(mod_extdisco, 1, OldModules),
+    ok = rpc(mim(), gen_mod_deps, replace_modules, [domain(), OldModules, NewModules]),
+    Config.
+
+request_external_services(To) ->
+    escalus_stanza:iq(To, <<"get">>,
+        [#xmlel{name = <<"services">>,
+                attrs = [{<<"xmlns">>, <<"urn:xmpp:extdisco:2">>}]}]).
+
+request_external_services_with_type(To, Type) ->
+    escalus_stanza:iq(To, <<"get">>,
+        [#xmlel{name = <<"services">>,
+                attrs = [{<<"xmlns">>, <<"urn:xmpp:extdisco:2">>},
+                         {<<"type">>, Type}]}]).
+
+request_external_services_credentials(To, Type, Host) ->
+    escalus_stanza:iq(To, <<"get">>,
```

Comment:
As ESL we have https://github.com/esl/escalus, so it would be kind of cool to move those stanzas there. I do not expect it in this PR, but can you make JIRA ticket for that, and later make a separate PR for that, please?

Reply:
Sure, I was thinking about that but didn't created a ticket for it. I will do it once this gets merged.